functions/New-DbaAgentProxy.ps1

```diff
@@ -108,7 +109,7 @@ function New-DbaAgentProxy {
         [string[]]$Name,
         [parameter(Mandatory)]
         [string[]]$ProxyCredential,
-        [ValidateSet("ActiveScripting", "AnalysisCommand", "AnalysisQuery", "CmdExec", "Distribution", "LogReader", "Merge", "PowerShell", "QueueReader", "Snapshot", "Ssis", "TransactSql")]
+        [ValidateSet("ActiveScripting", "AnalysisCommand", "AnalysisQuery", "CmdExec", "Distribution", "LogReader", "Merge", "PowerShell", "QueueReader", "Snapshot", "Ssis")]
```

Comment:
I was wondering why you removed the sub system "TransactSql" from the validate set on row 111? This is a valid option as you can see here: https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.agent.agentsubsystem?view=sql-smo-160

Reply:
When I was adding the integration tests for New-DbaAgentProxy I noticed that SMO throws an exception with that subsystem type:

```
Exception calling "AddSubSystem" with "1" argument(s): "Add subsystems failed for ProxyAccount 'dbatoolsci_proxytsql_68041246'. "
At <>\functions\New-DbaAgentProxy.ps1:212 char:25
+                         $proxy.AddSubSystem($system)
+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FailedOperationException
```

This is the underlying proc call that returns an error to SMO:

![tsql_subsystem_grant_statement_error](https://user-images.githubusercontent.com/68036988/113593945-e3678e80-95eb-11eb-920e-d9c0b19fefbf.png)